File: studyOC/studyOC/Scheme和Universal/SceneDelegate+Universal.m
//
//  SceneDelegate+Universal.m
//  studyOC
//
//  Created by Hahn on 2021/9/17.
//

#import "SceneDelegate+Universal.h"
#import "NSURL+Ex.h"
#import "LinkModel.h"

@implementation SceneDelegate (Universal)
- (void)scene:(UIScene *)scene continueUserActivity:(NSUserActivity *)userActivity API_AVAILABLE(ios(13.0))
{
  // 处理逻辑
    NSString *url = userActivity.webpageURL.host;
    NSDictionary *paras = [userActivity.webpageURL getURLParameters];
    NSLog(@"SceneDelegate universal url = %@",url);
    NSLog(@"SceneDelegate universal 参数 = %@", paras);
    
    LinkModel *model = [LinkModel new];
    model.url = url;
    model.para = paras;
    [[NSNotificationCenter defaultCenter] postNotificationName:@"Universal" object:model];
}
@end

File: studySwift/studySwift/Scheme和Universal/SceneDelegate+Universal.swift
//
//  SceneDelegate+Universal.swift
//  studySwift
//
//  Created by Hahn on 2021/9/17.
//

import UIKit

extension SceneDelegate {
    // 热启动 冷启动 调用
    func scene(_ scene: UIScene, continue userActivity: NSUserActivity) {
        
        DispatchQueue.main.asyncAfter(deadline: .now()+0.5, execute:{
            if let url = userActivity.webpageURL {
                let hostUrl = url.host
                let paras = url.absoluteURL.urlStringtoParams(url.absoluteString)
                print("SceneDelegate universal url = \(hostUrl)")
                print("SceneDelegate universal paras = \(paras)")
                
                var model = LinkModel()
                var para = Para()
                model.url = hostUrl
                para.name = paras["name"] as? String
                para.id = paras["id"] as? String
                para.age = paras["age"] as? String
                model.para = para
                
                NotificationCenter.default.post(name: NSNotification.Name.init(rawValue: "Universal"), object: model)
            }
        })
        
    }
}
